ember-addon-main.js

```diff
@@ -1,8 +1,92 @@
 'use strict';
 
+const fs = require('fs');
 const path = require('path');
 const utils = require('./utils');
 const hashForDep = require('hash-for-dep');
+const walkSync = require('walk-sync');
+const Plugin = require('broccoli-plugin');
+
+class ColocatedTemplateProcessor extends Plugin {
```

Comment:
FYI - This is absolutely not "ready for prime time", the implementation below is an initial spike (needs help with caching at the least).

Reply:
Created https://github.com/ember-cli/ember-cli-htmlbars/issues/290 to track this.